common/flatpak-dir.c

```diff
@@ -8909,6 +8909,188 @@ flatpak_dir_deploy_update (FlatpakDir        *self,
+      return;
+    }
+
+  /* Fix paths in desktop file with a find-and-replace. The portal handled
+   * quoting the app ID in the Exec line for us.
+   */
+  old_data = g_key_file_to_data (old_key_file, NULL, NULL);
+  data_string = g_string_new ((const char *)old_data);
+  g_string_replace (data_string, old_app_id, new_app_id, 0);
+  new_key_file = g_key_file_new ();
+  if (!g_key_file_load_from_data (new_key_file, data_string->str, -1,
+                                  G_KEY_FILE_KEEP_COMMENTS | G_KEY_FILE_KEEP_TRANSLATIONS,
+                                  &local_error))
+    {
+      g_warning ("Cannot load desktop file %s after rewrite: %s", desktop_path, local_error->message);
+      g_warning ("Key file contents:\n%s\n", (const char *)data_string->str);
+      return;
+    }
+  if (!g_key_file_save_to_file (new_key_file, desktop_path, &local_error))
+    {
+      g_warning ("Couldn't rewrite desktop file %s: %s", desktop_path, local_error->message);
+      return;
+    }
+
+  /* Rename desktop file */
+  desktop_suffix = desktop_name + strlen (old_app_id);
+  new_desktop = g_strconcat (new_app_id, desktop_suffix, NULL);
+  desktop_file = g_file_new_for_path (desktop_path);
+  if (!g_file_set_display_name (desktop_file, new_desktop, NULL, &local_error))
+    {
+      g_warning ("Unable to rename desktop file %s -> %s: %s", desktop_name, new_desktop, local_error->message);
+      return;
+    }
+
+  /* Fix symlink */
+  link_file = g_file_new_build_filename (g_get_user_data_dir (), "applications", desktop_name, NULL);
+  relative_path = g_build_filename ("..", "xdg-desktop-portal", "applications", new_desktop, NULL);
+  g_file_delete (link_file, NULL, NULL);
+  new_link_file = g_file_new_build_filename (g_get_user_data_dir (), "applications", new_desktop, NULL);
+  if (!g_file_make_symbolic_link (new_link_file, relative_path, NULL, &local_error))
+    {
+      g_warning ("Unable to rename desktop file link %s -> %s: %s", desktop_name, new_desktop, local_error->message);
+      return;
+    }
+
+  /* And rename the icon */
+  icon_path = g_key_file_get_string (old_key_file, G_KEY_FILE_DESKTOP_GROUP, "Icon", NULL);
+  if (g_str_has_prefix (icon_path, portal_icon_dir))
+    {
+      g_autoptr(GFile) icon_file = NULL;
+      g_autofree char *icon_basename = NULL;
+      g_autofree char *new_icon = NULL;
+      gchar *icon_suffix;
+
+      icon_file = g_file_new_for_path (icon_path);
+      icon_basename = g_path_get_basename (icon_path);
+      if (g_str_has_prefix (icon_basename, old_app_id))
+        {
+          icon_suffix = icon_basename + strlen (old_app_id);
+          new_icon = g_strconcat (new_app_id, icon_suffix, NULL);
+          if (!g_file_set_display_name (icon_file, new_icon, NULL, &local_error))
+            {
+              g_warning ("Unable to rename icon file %s -> %s: %s", icon_basename, new_icon,
+                         local_error->message);
```

Comment:
```suggestion
                         local_error->message);
              g_clear_error (&local_error);
```

To avoid the possibility of `local_error` being overwritten if more code is added below here in future and this is forgotten about.